mypy/build.py

```diff
@@ -2237,8 +2317,24 @@ def dispatch(sources: List[BuildSource], manager: BuildManager) -> Graph:
     # just want to load in all of the cache information.
     if manager.use_fine_grained_cache():
         process_fine_grained_cache_graph(graph, manager)
+        # Fine grained protocol dependencies are serialized separately, so we read them
+        # after we loaded cache for whole graph. The `read_protocol_cache` will also validate
+        # the protocol cache against the loaded individual cache files.
+        TypeState.proto_deps = read_protocol_cache(manager, graph)
+        if TypeState.proto_deps is None and manager.stats.get('fresh_trees', 0):
+            manager.errors.set_file(_get_prefix(manager), None)
+            manager.errors.report(0, 0, "Error reading protocol dependencies cache. Aborting.",
```

Comment:
Again, leave out the final period (maybe like this: "Error reading protocol dependencies cache -- aborting").